src/jlapi.c

```diff
@@ -310,6 +310,13 @@ DLLEXPORT jl_taggedvalue_t *jl_astaggedvalue(jl_value_t *v)
     return jl_astaggedvalue__MACRO(v);
 }
 
+// Create function versions of some useful macros
+#undef jl_valueof
+DLLEXPORT jl_value_t *jl_valueof(jl_taggedvalue_t *v)
+{
+    return jl_valueof__MACRO(v);
+}
```

Comment:
these would be easier to use if we just directly shadow the names (e.g. it might avoid the gdb `define macro jl_typeof jl_typeof` dance). I didn't do this initially because I was concerned some C parser out there would not handle these correctly:
```diff
diff --git a/src/jlapi.c b/src/jlapi.c
index a654d74..051d40f 100644
--- a/src/jlapi.c
+++ b/src/jlapi.c
@@ -304,16 +304,14 @@ DLLEXPORT const char *jl_git_commit()
 }
 
 // Create function versions of some useful macros
-#undef jl_astaggedvalue
-DLLEXPORT jl_taggedvalue_t *jl_astaggedvalue(jl_value_t *v)
+DLLEXPORT jl_taggedvalue_t *(jl_astaggedvalue)(jl_value_t *v)
 {
-    return jl_astaggedvalue__MACRO(v);
+    return jl_astaggedvalue(v);
 }
 
-#undef jl_typeof
-DLLEXPORT jl_value_t *jl_typeof(jl_value_t *v)
+DLLEXPORT jl_value_t *(jl_typeof)(jl_value_t *v)
 {
-    return jl_typeof__MACRO(v);
+    return jl_typeof(v);
 }
 
 #ifdef __cplusplus
diff --git a/src/julia.h b/src/julia.h
index 1d033e3..694eea5 100644
--- a/src/julia.h
+++ b/src/julia.h
@@ -172,10 +172,8 @@ typedef struct {
     jl_value_t value;
 } jl_taggedvalue_t;
 
-#define jl_astaggedvalue__MACRO(v) container_of((v),jl_taggedvalue_t,value)
-#define jl_typeof__MACRO(v) ((jl_value_t*)(jl_astaggedvalue__MACRO(v)->type_bits&~(uintptr_t)15))
-#define jl_astaggedvalue jl_astaggedvalue__MACRO
-#define jl_typeof jl_typeof__MACRO
+#define jl_astaggedvalue(v) container_of((v),jl_taggedvalue_t,value)
+#define jl_typeof(v) ((jl_value_t*)(jl_astaggedvalue(v)->type_bits&~(uintptr_t)15))
 static inline void jl_set_typeof(void *v, void *t)
 {
     jl_taggedvalue_t *tag = jl_astaggedvalue(v);
```